instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -202,19 +248,21 @@ def get(
         if not headers:
             return None
 
-        # asgi header keys are in lower case
+        # ASGI header keys are in lower case
         key = key.lower()
         decoded = [
             _value.decode("utf8")
             for (_key, _value) in headers
-            if _key.decode("utf8") == key
+            if _key.decode("utf8").lower() == key
         ]
         if not decoded:
             return None
         return decoded
 
     def keys(self, carrier: dict) -> typing.List[str]:
-        return list(carrier.keys())
+        return [
+            _key.decode("utf8") for (_key, _value) in carrier.get("headers")
```

Comment:
The `carrier` here is expected to be headers dict so the `carrier.keys()` should be fine. If the caller is incorrectly using then I think fix should be implemented there.

Reply:
Ok, I left the decode() in but I took the other part out.  Let me know if you want that part removed too.